cmd/serverless-init/metric/metric_test.go

```diff
@@ -64,6 +64,17 @@ func TestAddShutdownMetric(t *testing.T) {
 	assert.Equal(t, metric.Tags[1], "tagb:valueb")
 }
 
+func TestNilDemuxDoesNotPanic(t *testing.T) {
+	demux := createDemultiplexer(t)
+	timestamp := time.Now()
+	// Pass nil for demux to mimic when a port is blocked and dogstatsd does not start properly.
+	// This previously led to a panic and segmentation fault
+	AddColdStartMetric("gcp.run", []string{"taga:valuea", "tagb:valueb"}, timestamp, nil)
+	generatedMetrics, timedMetrics := demux.WaitForSamples(100 * time.Millisecond)
+	assert.Equal(t, 0, len(timedMetrics))
+	assert.Equal(t, 0, len(generatedMetrics))
+}
```

Comment:
If you decide to move the nil check to the `add` function, then you should update this test to call and test the `add` function instead of `AddColdStartMetric`. Otherwise, you should make sure to test both the `AddColdStartMetric` and `AddShutdownMetric`.